### PR TITLE
Update to 0.9 syntax and incorporate CTAN feedback.

### PR DESCRIPTION
**Pull request by cgranade**
This PR improves qsharp.sty by including feedback from the 0.3 CTAN submission, and by updating the language definition to include new features from Q# 0.4 through to 0.9.

### CODE REVIEW

File: qsharp.dtx [hunk 1]
```diff
@@ -151,23 +187,62 @@
...
         function,
         body,
         adjoint,
         auto,
+        distribute,
+        self,
+        invert,
         controlled,
         let,
         set,
```
Comment (bettinaheim) [inline qsharp.dtx:203]:
```suggestion
        set,
        if, 
        elif, 
        else,
        repeat, 
        until,
        fixup,
        while,
        return, 
        fail,
```

File: qsharp.dtx [hunk 2]
```diff
@@ -151,23 +187,62 @@
...
         borrowing,
+        within,
+        apply,
+        is,
+        Adj,
+        Ctl,
         Adjoint,
+        Controlled,
         Qubit, Unit,
         Zero, One
```
Comment (bettinaheim) [inline qsharp.dtx]:
```suggestion
        Zero, One,
        true, false, 
        PauliI, PauliX, PauliY, PauliZ,
        new, 
        not, and, or,        
```

File: qsharp.dtx [hunk 3]
```diff
@@ -151,23 +187,62 @@
     morekeywords={
         namespace,
         open,
+        as,
         operation,
         function,
```
Comment (bettinaheim) [inline qsharp.dtx:192]:
```suggestion
        function,
        newtype,
```

File: qsharp.dtx [hunk 4]
```diff
@@ -151,23 +187,62 @@
...
         body,
         adjoint,
         auto,
+        distribute,
+        self,
+        invert,
```
Comment (bettinaheim) [inline qsharp.dtx:199]:
```suggestion
        invert,
        intrinsic,
```

File: qsharp.dtx [hunk 5]
```diff
@@ -151,23 +187,62 @@
...
         using,
         borrowing,
+        within,
+        apply,
+        is,
+        Adj,
+        Ctl,
         Adjoint,
+        Controlled,
         Qubit, Unit,
```
Comment (bettinaheim) [inline qsharp.dtx:225]:
```suggestion
        Qubit, Unit,
        Int, BigInt,
        Double, 
        Bool, 
        String, 
        Pauli, Result,
        Range,
```

**Comment by cgranade** [comment]
Thanks for the help, @bettinaheim!

**Review by bettinaheim — APPROVED** [review]

**Pull request merged**